EasyClangComplete.py

```diff
@@ -78,6 +78,21 @@ def run(self, edit):
             log.debug(" Nothing to clean")
 
 
+class ApplyFixitCommand(sublime_plugin.TextCommand):
+    """Command to apply fixit at cursor position."""
+
+    def run(self, edit):
+        """All the actual processing is done in on_text_command."""
+
+
+class ReplaceCharactersCommand(sublime_plugin.TextCommand):
+    """Command to actually replace region with fixit value."""
+
+    def run(self, edit, region, value):
+        """Replace given region with given string."""
+        self.view.replace(edit, sublime.Region(*region), value)
```

Comment:
If I got it right this is the way the Sublime API works. To replace a substring you need an Edit object (https://www.sublimetext.com/docs/3/api_reference.html#sublime.Edit) which you get when a TextCommand is processed. The `show_popup_menu` call is asynchronous, that is, we don't know when its `on_done` argument (`process_fixit` in this case) will be called. So I can't reuse the Edit object from the ApplyFixit command and need to run another command to get a new Edit object.

I didn't find a default command for text replacement (is there an exhaustive list of sublime text commands?) so I had to add this class.